networkx/algorithms/approximation/dominating_set.py

```diff
@@ -21,71 +21,96 @@
 #   Nicholas Mancuso <nick.mancuso@gmail.com>
 #   All rights reserved.
 #   BSD license.
-import networkx as nx
+from __future__ import division
+
+from ..matching import maximal_matching
+from ...utils import not_implemented_for
+
 __all__ = ["min_weighted_dominating_set",
            "min_edge_dominating_set"]
+
 __author__ = """Nicholas Mancuso (nick.mancuso@gmail.com)"""
 
 
+# TODO Why doesn't this algorithm work for directed graphs?
```

Comment:
I suspect it is because how you define dominating in a directed graph may depend on the direction of the edge. Perhaps you want dominating to me that there is an outgoing edge from each node in the dominating set to every node that is not in the dominating set.